libs/layout/src/lib/directives/display-content/display-content.directive.ts

```diff
@@ -0,0 +1,199 @@
+import {
+	AfterViewInit,
+	ChangeDetectorRef,
+	Directive,
+	Inject,
+	Input,
+	OnDestroy,
+	TemplateRef,
+	Type,
+	ViewContainerRef,
+} from '@angular/core';
+import { Subject, distinctUntilChanged, takeUntil, tap } from 'rxjs';
+import {
+	NgxDisplayContentConditions,
+	NgxDisplayContentConfiguration,
+	NgxDisplayContentOverrideConfiguration,
+	NgxDisplayContentStatus,
+} from '../../types';
+import { NgxDisplayContentConfigurationToken } from '../../tokens';
+import { NgxOnlineService } from '../../services';
+import { NgxDisplayContentComponent } from '../../abstracts';
+
+/**
+ * NgxDisplayContentDirective handles whether content should be shown based on a loading, error or offline condition by using provided default components or a provided override template
+ */
+@Directive({
+	selector: '[displayContent]',
+	standalone: true,
+})
+export class NgxDisplayContentDirective implements AfterViewInit, OnDestroy {
+	/**
+	 * A subject to handle the destroyed flow
+	 */
+	private readonly onDestroySubject: Subject<void> = new Subject<void>();
+
+	/**
+	 * A subject to know when we need to update the view
+	 */
+	private readonly updateViewSubject: Subject<void> = new Subject<void>();
+
+	/**
+	 * Holds an optional override configuration
+	 */
+	private overrideConfiguration: NgxDisplayContentOverrideConfiguration = {};
+
+	/**
+	 * Holds the conditions we use to render the component
+	 */
+	private conditions: NgxDisplayContentConditions = {};
+
+	/**
+	 * Renders the item or a default fallback based on the provided conditions
+	 */
+	@Input() set displayContent(conditions: NgxDisplayContentConditions) {
+		// Iben: Update the conditions, but use a merge so that the online status can be preserved
```

Comment:
I would move this to a private method and re-use it:
```typescript
// Iben: Update the conditions, but use a merge so that the online status can be preserved
if (conditions) {
	this.conditions = {
		...this.conditions,
		...conditions,
	};
}

// Iben: Notify that the view needs to be updated
this.updateViewSubject.next();
```